src/handle_form.inl

```diff
@@ -701,43 +705,71 @@ mg_handle_form_request(struct mg_connection *conn,
 				return -1;
 			}
 
+			/* @see https://www.rfc-editor.org/rfc/rfc2046.html#section-5.1.1
+			 *
+			 * multipart-body := [preamble CRLF]
+			 *     dash-boundary transport-padding CRLF
+			 *     body-part *encapsulation
+			 *     close-delimiter transport-padding
+			 *     [CRLF epilogue]
+			 */
+
 			if (part_no == 0) {
-				int d = 0;
-				while ((d < buf_fill) && (buf[d] != '-')) {
-					d++;
+				size_t preamble_length = 0;
+				/* skip over the preamble until we find a complete boundary */
+				/* +2 for the -- preceding the boundary */
+				while ((preamble_length < buf_fill - bl)
```

Comment:
Could drop this / add a preamble length limit if we're worried about legitimate request sizes

Reply:
I have no idea why someone would want to use a preamble at all.
A preamble >4kB seems excessive and is probably only sent to block the server.

Reply:
Done - went for 1024